### PR TITLE
feat: add release checklist

### PR DESCRIPTION
**Pull request by ServiceDog**

### CODE REVIEW

File: docs/contributing/guides/release-checklist.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  - [ ] Bibliography (if any citations)
+- [ ] README (see svc-cargo for example)
+  - [ ] Badges
+  - [ ] Accurate installation instructions
+  - [ ] Current links to other documents
+  - [ ] Arrow DAO links
+
+## Code (not applicable to examples/ folder)
+- [ ] No unhandled unwrap()s
+- [ ] No ()?; on functions with several exits
```
Comment (romanmandryk) [inline docs/contributing/guides/release-checklist.md:65]: Can you explain reasoning here? Why not use `?` specifically in functions with several exits?

I would say in public functions (library) or request handlers(service) we should "catch" the error, log it and then return it back to user/caller (rethrow). In all inner functions `?` (error propagation) is actually good to use as much as we can? ( within Rust limitations)

**Review by romanmandryk — DISMISSED** [review]

File: docs/contributing/guides/release-checklist.md (hunk 1 above)
Comment (ServiceDog) [reply docs/contributing/guides/release-checklist.md:65]: I don't see any Rust RFC's that state it shouldn't be used in production though, so I would be fine removing this bullet

Comment (ServiceDog) [reply docs/contributing/guides/release-checklist.md:65]: To be clear, I mean this sort of setup:
```rust
fn do_something() -> Result<(), Box<dyn Error>> {
    fxn_1()?;
    // ... 
    fxn_2()?;
    // ...
    fxn_3()?;
}
```

**Review by GoodluckH — APPROVED** [review]

File: docs/contributing/guides/release-checklist.md (hunk 1 above)
Comment (ServiceDog) [reply docs/contributing/guides/release-checklist.md]: Removed this bullet

**Review by GoodluckH — APPROVED** [review]

**Comment by ServiceDog** [comment]
@owlot Look good? We'll be doing this process this week

**Review by GoodluckH — COMMENTED** [review]
Do we need to include contributor's github username in the branch name? i.e., `goodluckh/r1-final-review`

**Comment by ServiceDog** [comment]
> Do we need to include contributor's github username in the branch name? i.e., `goodluckh/r1-final-review`

There should only ever be one r1-final-review branch, not critical for there to be the github username imo

**Review by romanmandryk — APPROVED** [review]

**Review by romanmandryk — APPROVED** [review]

**Review by GoodluckH — APPROVED** [review]

**Review by owlot — APPROVED** [review]

**Pull request merged**